File: src/mod10_design_pattern/lesson2_factory/java/MagicalCreatureFactory.java
package mod10_design_pattern.lesson2_factory.java;

public class MagicalCreatureFactory {

    public static MagicalCreature createCreature(String type) {
        if (type.equalsIgnoreCase("griffin")){
            return new Griffin();
        }else if (type.equalsIgnoreCase("gragon")){
            return new Dragon();
        }else {
            throw new IllegalArgumentException(type + "is not exist!");
        }
    }
}

File: src/mod10_design_pattern/lesson2_factory/kotlin/MagicalCreatureFactory.kt
package mod10_design_pattern.lesson2_factory.kotlin

object MagicalCreatureFactory {
    fun createCteature(type: String): MagicalCreature {
        return when (type.lowercase()) {
            "griffin" -> Griffin()
            "dragon" -> Gragon()
            else -> throw IllegalArgumentException("type $type is incorrect type!")
        }
    }
}